src/lib/notify.js

```diff
@@ -223,3 +236,34 @@ Notification.prototype.defaultOptions = {
   },
   persistence: 1
 };
+
+
+export default class NotificationWatcher {
+  constructor() {
+    if (!storage.nextPrompt) {
+      storage.nextPrompt = this.nextPrompt();
+    }
+    this.intervalCallback();
+    this.interval = setInterval(() => this.intervalCallback(), POLL_INTERVAL);
+  }
+
+  nextPrompt() {
+    return new Date().getTime() + PROMPT_INTERVAL;
+  }
+
+  intervalCallback() {
+    const now = new Date().getTime();
+    if (now < storage.nextPrompt) {
+      logger.log(`Next prompt in ${(storage.nextPrompt - now)/1000}s`);
+      return;
+    }
+    logger.log(`Setting up prompt, will show in ${POLL_INTERVAL/1000}s`);
+    this.timeout = setTimeout(() => this.showPrompt(), POLL_INTERVAL);
```

Comment:
Why set up another timer, rather than just show the prompt immediately here since the interval has elapsed? Works, but seems like an extra step.